query/iterator.gen.go

```diff
@@ -411,14 +417,48 @@ func (h *floatSortedMergeHeap) Less(i, j int) bool {
 		} else if xTags, yTags := x.Tags.Subset(h.opt.Dimensions), y.Tags.Subset(h.opt.Dimensions); !xTags.Equals(&yTags) {
 			return xTags.ID() < yTags.ID()
 		}
-		return x.Time < y.Time
+
+		if x.Time != y.Time {
+			return x.Time < y.Time
+		}
+
+		if len(x.Aux) > 0 && len(x.Aux) == len(y.Aux) {
+			for i := 0; i < len(x.Aux); i++ {
+				v1, ok1 := x.Aux[i].(string)
+				v2, ok2 := y.Aux[i].(string)
+				if !ok1 || !ok2 {
+					// Unsupported types used in Aux fields. Maybe they
+					// need to be added here?
+					return false
+				} else if v1 == v2 {
+					continue
+				}
+				return v1 < v2
+			}
+		}
```

Comment:
Given the `<type>Point#Aux` field is populated, assuming the length of the original `IteratorOptions#Aux` field, [here][0], therefore I would argue this is currently an invariant. Additionally, the only place the `aux` field of the iterator is built is via [buildVarRefSeriesIterator][1], again derived from the `IteratorOptions#Aux` field

[0]: https://github.com/influxdata/influxdb/blob/d189621d072a6fc3224cfb46c52855ac085d5317/tsdb/engine/tsm1/iterator.gen.go.tmpl#L220-L222
[1]: https://github.com/influxdata/influxdb/blob/d189621d072a6fc3224cfb46c52855ac085d5317/tsdb/engine/tsm1/engine.go#L1853